cirq/protocols/channel.py

```diff
@@ -130,3 +143,38 @@ def channel(val: Any,
     raise TypeError("object of type '{}' does have a _channel_, _mixture_ or "
                     "_unitary_ method, but it returned NotImplemented."
                     .format(type(val)))
+
+
+def has_channel(val: Any) -> bool:
+    """Returns whether the value has a channel representation.
+
+    Returns:
+        If `val` has a `_has_channel_` method and its result is not
+        NotImplemented, that result is returned. Otherwise, if `val` has a
+        `_has_mixture_` method and its result is not NotImplemented, that
+        result is returned. Otherwise if `val` has a `_has_unitary_` method
+        and its results is not NotImplemented, that result is returned.
+        Otherwise, if the value has a _channel_ method return if that
+        has a non-default value. Returns False if none of these functions
+        exists.
+    """
+    channel_getter = getattr(val, '_has_channel_', None)
+    result = NotImplemented if channel_getter is None else channel_getter()
+
+    if result is not NotImplemented:
+        return result
+
+    mixture_getter = getattr(val, '_has_mixture_', None)
+    result = NotImplemented if mixture_getter is None else mixture_getter()
+
+    if result is not NotImplemented:
```

Comment:
I think we should also check for `None`. This has been leaking into some of the other protocols; users are more familiar with it I think. It makes sense when the None result would clearly be invalid.